etcdserver/raft.go

```diff
@@ -97,16 +98,25 @@ type raftNode struct {
 	// last lead elected time
 	lt time.Time
 
+	// to check if msg receiver is removed from cluster
+	isIDRemoved func(id uint64) bool
```

Comment:
i feel it is OK to have a cluster reference in raft struct. it is better than this func pointer in my opinion.